core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -1025,7 +1025,7 @@ func (ec *EthConfirmer) markForRebroadcast(etx EthTx, head *evmtypes.Head) error
 
 	// Rebroadcast the one with the highest gas price
 	attempt := etx.EthTxAttempts[0]
-	var receipt EthReceipt
+	var receipt txmgrtypes.Receipt[evmtypes.Receipt, gethCommon.Hash]
```

Comment:
We could limit a lot of noise by defining a type (or alias?) for this combination of generic params:
```go
type EthReceipt txmgrtypes.Receipt[evmtypes.Receipt, gethCommon.Hash]
type EthReceipt = txmgrtypes.Receipt[evmtypes.Receipt, gethCommon.Hash]
```

Reply:
nit: Prefer EvmReceipt than EthReceipt.

Reply:
We already have a Receipt (the raw receipt) defined under evm/types/types.go so it could be confusing. Since this generic struct is only used once here let's keep as is.

Reply:
Aren't there around a dozen of these repeated? Why not just redefine the old type as this new generic one? (or rename from `EthReceipt` to `EVMReceipt` :shrug:)

Reply:
gotcha, good point. We'll add that to the backlog (changing legacy naming)